src/App.tsx

```diff
@@ -15,36 +16,113 @@
   'Garlic',
 ];
 
+enum SortType {
+  alphabetically = 'alphabetically',
+  byLength = 'length',
+  default = '',
+}
+
+interface SortParams {
+  sortField: SortType;
+  isReversed: boolean;
+}
+
+function getPrepereGoods(
+  goods: string[],
+  { sortField, isReversed }: SortParams,
+) {
+  let prepareGoods = [...goods];
+
+  if (sortField === SortType.alphabetically) {
+    prepareGoods.sort((good1, good2) => good1.localeCompare(good2));
+  }
+
+  if (sortField === SortType.byLength) {
+    prepareGoods.sort((good1, good2) => good1.length - good2.length);
+  }
+
+  if (isReversed) {
+    prepareGoods = prepareGoods.reverse();
+  }
+
+  return prepareGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.default);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleGoods = getPrepereGoods(goodsFromServer, {
```

Comment:
Ensure that the function name `getPrepereGoods` is corrected to `getPrepareGoods` here as well, to match the corrected function name.